internal/examples/agent/agent/agent.go

```diff
@@ -393,6 +487,15 @@ func (agent *Agent) onMessage(ctx context.Context, msg *types.MessageData) {
 			agent.logger.Errorf(err.Error())
 		}
 	}
+
+	// TODO: check that the Server has AcceptsConnectionSettingsRequest capability before
```

Comment:
Can this be part of the Client, which keeps track of `ServerCapabilities` and returns an error when `RequestConnectionSettings` is called?

Reply:
It is a possible option. The difficulty is that initially the Client does not know about server capabilities until the very first message from the Server arrives. We need to find a good way to expose this in the Client API.